### PR TITLE
Commando: "Somewhere, somehow, someone's going to pay."

### PR DESCRIPTION
**Pull request by rustyrussell**
This is, honestly, the greatest plugin ever written.  

It allows use of the already-established Lightning network protocol to control a node, and in addition supports authorization cookies (called "runes") which are easy to craft and powerful.  Runes can restrict what commands can be run, what nodes can run them, what parameters can be given, limit what time the rune is valid for, and ratelimit how often the rune can be used.  You can decode them into human readable form, and add new restrictions to them.

This is a reimplementation (and streamlining) of the `commando.py` plugin.  It is always on, but doesn't do anything until you create your first rune.

It is fully documented, too.  With examples!

Credit for naming: @cdecker

### CODE REVIEW

File: doc/schemas/commando-rune.schema.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "additionalProperties": true,
```
Comment (SimonVrouwe) [inline doc/schemas/commando-rune.schema.json:4]: Can be `false`?  No extra stuff is ever added it seems.

Comment (rustyrussell) [reply doc/schemas/commando-rune.schema.json:4]: Indeed, wonder where I cut & paste that from?

**Comment by rustyrussell** [comment]
Rebased on #5735:

- Fixed schema flaw found by @SimonVrouwe 
- Added typo fix from @zerofeerouting

**Comment by jb55** [comment]
this is amazing! testing now

**Comment by rustyrussell** [comment]
Merged typo fix by @jb55 and rebased on master (again!) to fix some flakes...

**Comment by rustyrussell** [comment]
Rebase on master, thanks to some prereqs being merged from @adi2011 !

**Comment by rustyrussell** [comment]
Needs rebase now that makesharesecret is in with different API.  See #5430

**Comment by rustyrussell** [comment]
Rebased on #5430

**Comment by jb55** [comment]
I turned off the plugin and ran this branch and all my apps still work. amazing.

**Comment by rustyrussell** [comment]
> and it's fast 🤯
> 
> looks like guilt got confused and made me the author on [74785ce](https://github.com/ElementsProject/lightning/commit/74785ce5a98d21e66b6dce461431325694d9002b) ?

Ah, I squashed your fix in there. I'll fix, especially since I have to rebase to fix conflicts anyway...

**Comment by rustyrussell** [comment]
Rebased on master, fixed @jb55 fix which made him author of the plugin :)

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/commit/2e13b72f55080be07ea68de77976eb990a043f5d

The following test failure looks unrelated, and also looks like untracked? cc @niftynei

```
=================================== FAILURES ===================================
__________________________ test_v2_open_sigs_restart ___________________________
[gw0] linux -- Python 3.7.13 /opt/hostedtoolcache/Python/3.7.13/x64/bin/python3
node_factory = <pyln.testing.utils.NodeFactory object at 0x7f5e8b394d10>
bitcoind = <pyln.testing.utils.BitcoinD object at 0x7f5e8ac14490>
@unittest.skipIf(TEST_NETWORK != 'regtest', 'elementsd doesnt yet support PSBT features we need')
@pytest.mark.developer("uses dev-disconnect")
@pytest.mark.openchannel('v2')
deftest_v2_open_sigs_restart(node_factory, bitcoind):
        disconnects_1 = ['-WIRE_TX_SIGNATURES']
        disconnects_2 = ['+WIRE_TX_SIGNATURES']
        l1, l2 = node_factory.get_nodes(2,
                                        opts=[{'disconnect': disconnects_1,
'may_reconnect': True},
                                              {'disconnect': disconnects_2,
'may_reconnect': True}])
        l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
        amount = 2**24
        chan_amount = 100000
        bitcoind.rpc.sendtoaddress(l1.rpc.newaddr()['bech32'], amount / 10**8 + 0.01)
        bitcoind.generate_block(1)
# Wait for it to arrive.
        wait_for(lambda: len(l1.rpc.listfunds()['outputs']) > 0)
# Fund the channel, should appear to finish ok even though the
# peer fails
with pytest.raises(RpcError):
            l1.rpc.fundchannel(l2.info['id'], chan_amount)
        chan_id = first_channel_id(l1, l2)
        log = l1.daemon.is_in_log('{} psbt'.format(chan_id))
assert log
        psbt = re.search("psbt (.*)", log).group(1)
        l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
        l1.daemon.wait_for_log('Peer has reconnected, state DUALOPEND_OPEN_INIT')
with pytest.raises(RpcError):
>           l1.rpc.openchannel_signed(chan_id, psbt)
E           Failed: DID NOT RAISE <class 'pyln.client.lightning.RpcError'>

```

**Pull request merged**